### PR TITLE
Create a first class structure

### PR DESCRIPTION
**Pull request by orionsamuel**

### CODE REVIEW

File: deck.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include "deck.hpp"
+#include <string>
+#include <memory>
+#include <algorithm>
+#include <random>
+#include <chrono>
+
+Deck::Deck() {
+    std::vector<std::string> suits = {"Spades", "Hearts", "Clubs", "Diamonds"};
+    std::vector<std::string> values = {"Ace", "Two", "Three", "Four", "Five", "Six", "Seven", "Jack", "Queen", "King"};
```
Comment (henriquefl) [inline deck.cpp]: Better to create a Enum?

File: card.cpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#include "card.hpp"
+#include <iostream>
+
+Card::Card(std::string s, std::string v) : suit(s), value(v) {}
+
+void Card::displayCard() {
```
Comment (henriquefl) [inline card.cpp:23]: Maybe we should move the in/out operations to outside the model? Better to create the getters and setters and leave this responsibility to the screen.

File: gameController.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#include "gameController.hpp"
+#include <fstream>
+#include <iostream>
+
+namespace fs = std::filesystem;
+
+GameController::GameController() {
+    gameDeck = std::make_shared<Deck>();
+    player1 = std::make_shared<Player>();
```
Comment (henriquefl) [inline gameController.cpp]: Maybe we should have a list of players instead of make a fixed number.

File: gameController.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    //Game turn rules
+    currentPlayer->setScore();
+
+    //Define stop condition
+    if(player1->getScore() >= maxScore || player2->getScore() >= maxScore){
+        std::cout << "Game Over! Final scores - Player 1: " << player1->getScore() << " | Player 2: " << player2->getScore() << std::endl;
+        return;
+    }
+
+    int nextPlayerNumber = (currentPlayerNumber == 1) ? 2 : 1;
```
Comment (henriquefl) [inline gameController.cpp]: Considering multiple players, maybe we can consider a circular list.

File: gameController.cpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    std::shared_ptr<Player> nextPlayer = (nextPlayerNumber == 1) ? player1 : player2;
+
+    saveGame("game_state.txt");
+    playRound(nextPlayer, nextPlayerNumber);
+
+}
+
+void GameController::saveGame(const std::string& filename) {
+    try {
```
Comment (henriquefl) [inline gameController.cpp:52]: Can we move this logic to outside the game controller?

File: card.cpp (hunk 2 above)
Comment (orionsamuel) [reply card.cpp:23]: Done

File: deck.cpp (hunk 1 above)
Comment (orionsamuel) [reply deck.cpp]: Done

File: gameController.cpp (hunk 5 above)
Comment (orionsamuel) [reply gameController.cpp:52]: Do you have any sugestion?

**Pull request merged**